contracts/lockdrop/src/contract.rs

```diff
@@ -563,11 +575,29 @@ fn migrate_pair_step_1(
         .as_ref()
         .ok_or_else(|| StdError::generic_err("Generator address hasn't set yet!"))?;
 
+    let new_pool_info: PoolInfoV2 = ASSET_POOLS_V2.load(deps.storage, pool_type)?;
+    let new_pool_addr = get_lp_token_pool_addr(deps.as_ref(), &new_pool_info.lp_token)?;
+    let new_pool_info: astroport::pair::PoolResponse = deps
+        .querier
+        .query_wasm_smart(new_pool_addr, &astroport::pair::QueryMsg::Pool {})?;
     //get current ntrn contract balance
     let current_ntrn_balance = deps
```

Comment:
I don't really like how the `current_ntrn_balance` querying is put in the middle of `current_token_balance` calculation, that's distracting. please separate these two processes somehow